server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java

```diff
@@ -42,23 +43,31 @@
   private final DbClient dbClient;
   private final IssueStorage issueStorage;
   private final NotificationManager notificationService;
+  private final LiveMeasureComputer liveMeasureComputer;
 
-  public IssueUpdater(DbClient dbClient, IssueStorage issueStorage, NotificationManager notificationService) {
+  public IssueUpdater(DbClient dbClient, IssueStorage issueStorage, NotificationManager notificationService,
+    LiveMeasureComputer liveMeasureComputer) {
     this.dbClient = dbClient;
     this.issueStorage = issueStorage;
     this.notificationService = notificationService;
+    this.liveMeasureComputer = liveMeasureComputer;
   }
 
   /**
    * Same as {@link #saveIssue(DbSession, DefaultIssue, IssueChangeContext, String)} but populates the specified
    * {@link SearchResponseData} with the DTOs (rule and components) retrieved from DB to save the issue.
    */
-  public SearchResponseData saveIssueAndPreloadSearchResponseData(DbSession session, DefaultIssue issue, IssueChangeContext context, @Nullable String comment) {
+  public SearchResponseData saveIssueAndPreloadSearchResponseData(DbSession session, DefaultIssue issue, IssueChangeContext context,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "refreshMeasures" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2301)